infra/modules/azure_function_app_exposed/locals.tf

```diff
@@ -4,16 +4,22 @@ locals {
   domain          = var.environment.domain == null ? "-" : "-${var.environment.domain}-"
   app_name_prefix = "${local.project}${local.domain}${var.environment.app_name}"
 
+  # For backwards compatibility
+  # If no legacy value is passed, use the original value
+  # TO DO: Remove this in the next major release and replace all local.tier with var.tier
```

Comment:
every "todo" must have a reference to some jira issue (id), otherwise they become _neverdo_